BoxDecorationPainter should not use InterpolationQuality::Default with ImagePaintingOptions

kkinnunen-apple opened the pull request:
#### 78d90a8e53b2f787f468ac358bb024e4f7390ef2
<pre>
BoxDecorationPainter should not use InterpolationQuality::Default with ImagePaintingOptions
<a href="https://bugs.webkit.org/show_bug.cgi?id=262360">https://bugs.webkit.org/show_bug.cgi?id=262360</a>
rdar://116228654

Reviewed by Tim Nguyen.

Passing &quot;Default&quot; in ImagePaintingOptions means &quot;Use what is in
GraphicsContext&quot;. It is the same as not passing anything.

This is work towards removing ImagePaintingOptions::Default, as it
does not mean what the developers think it means (platform default
image quality).

* Source/WebCore/display/css/DisplayBoxDecorationPainter.cpp:
(WebCore::Display::BoxDecorationPainter::paintFillLayer const):

Canonical link: <a href="https://commits.webkit.org/268650@main">https://commits.webkit.org/268650@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/19cd56e2e2fa7047d1ce2f312b40c63a00c7420f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20232 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20664 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21302 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22132 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18881 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20463 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23917 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20831 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20326 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20451 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20358 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17589 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22982 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17526 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18392 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24652 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18603 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18568 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22624 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19149 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16260 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18354 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4873 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22696 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18974 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/display/css/DisplayBoxDecorationPainter.cpp:
```diff
@@ -1252,8 +1252,7 @@ void BoxDecorationPainter::paintFillLayer(PaintingContext& paintingContext, cons
         layer.blendMode(),
         DecodingMode::Synchronous,
-        ImageOrientation::Orientation::FromImage,
-        InterpolationQuality::Default
```
I wonder if it should use this method: https://searchfox.org/wubkat/source/Source/WebCore/rendering/RenderBoxModelObject.cpp#659 like `BackgroundPainter::paintFillLayer` does.

The code in display/ is dead code as we speak though, so it doesn't matter too much.

kkinnunen-apple committed:
BoxDecorationPainter should not use InterpolationQuality::Default with ImagePaintingOptions
https://bugs.webkit.org/show_bug.cgi?id=262360
rdar://116228654

Reviewed by Tim Nguyen.

Passing "Default" in ImagePaintingOptions means "Use what is in
GraphicsContext". It is the same as not passing anything.

This is work towards removing ImagePaintingOptions::Default, as it
does not mean what the developers think it means (platform default
image quality).

* Source/WebCore/display/css/DisplayBoxDecorationPainter.cpp:
(WebCore::Display::BoxDecorationPainter::paintFillLayer const):

Canonical link: https://commits.webkit.org/268650@main
